### PR TITLE
[SYCLomatic][part3]Refine cmake migration framework to support full word match and global build-in rule file

### PR DESCRIPTION
**Pull request by tomflinda**
1 Enable full word match for cmake syntax migration rule
2 Skip user defined migration run for some cmake syntax if implicit migration rule is available
3 Use a global build-in built-in migration rule file for all the lit tests
4 Split a separate analysis parse to collect variable and its value before applying migration rules

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/MigrateCmakeScript.cpp [hunk 1]
```diff
@@ -501,18 +461,51 @@ int skipCmakeComments(const std::string &Input, int Index) {
...
+  Value = Input.substr(Begin, End - Begin);
+
+  std::string ReplStr = processArgOfCmakeVersionRequired(Value, CmakeVarMap);
+
+  Input.replace(Begin, End - Begin, ReplStr);
+  Size = Input.size();            // Update string size
+  Index = Begin + ReplStr.size(); // Update index
+}
+
+void applyImplicitMigrationRules(std::string &Input) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrateCmakeScript.cpp:502]: Please add comment on: implict migration Rules is used in what kind of scenarsio.

File: clang/lib/DPCT/MigrateCmakeScript.cpp [hunk 2]
```diff
@@ -604,3 +566,93 @@ std::string convertCmakeCommandsToLower(const std::string &InputString) {
...
+}
+
+static void applyCmakeMigrationRules() {
+  for (auto &Entry : CmakeScriptFileBufferMap) {
+    auto &Buffer = Entry.second;
+
+    // Apply implicit migration rules
+    applyImplicitMigrationRules(Buffer);
+
+    // Apply user define migration rules
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrateCmakeScript.cpp:603]: please comment on:  when yaml based rule is used,  or used for what kind cases.

Comment (tomflinda) [reply clang/lib/DPCT/MigrateCmakeScript.cpp:603]: ImplicitMigrationRules is used when the migration logic is difficult to be described with yaml based rule syntax. 
Currently, only migration of cmake_minimum_required() is applied with ImplicitMigrationRule. Once ImplicitMigrationRule for some cmake syntax is implemented, the yaml based rules for the same cmake syntax will be skipped and a msg is emitted to notify the user.

File: clang/lib/DPCT/MigrateCmakeScript.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/MigrateCmakeScript.cpp:502]: ImplicitMigrationRules is used when the migration logic is difficult to be described with yaml based rule syntax.
Currently, only migration of cmake_minimum_required() is applied with ImplicitMigrationRule. Once ImplicitMigrationRule for some cmake syntax is implemented, the yaml based rules for the same cmake syntax will be skipped and a msg is emitted to notify the user.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**